src/controller/CHIPDeviceController.cpp

```diff
@@ -1209,192 +1143,167 @@ CHIP_ERROR DeviceCommissioner::ValidateAttestationInfo(const ByteSpan & attestat
                                                &mDeviceAttestationInformationVerificationCallback);
 
     // TODO: Validate Firmware Information
-
-    return CHIP_NO_ERROR;
 }
 
-void DeviceCommissioner::HandleAttestationResult(CHIP_ERROR err)
+TRAMPOLINE(DeviceAttestationInformationVerification, AttestationVerificationResult);
+void DeviceCommissioner::OnDeviceAttestationInformationVerification(AttestationVerificationResult result)
 {
-    if (err != CHIP_NO_ERROR)
+    StopSessionIf(mState != State::Initialized, CHIP_ERROR_INCORRECT_STATE);
+    StopSessionIf(mDeviceBeingCommissioned == nullptr, CHIP_ERROR_INCORRECT_STATE);
```

Comment:
These checks did not use to be in `OnDeviceAttestationInformationVerification`.  Why are they being added?  Is it because we used to cancel the callback when either of those things became true and now we no longer use cancelable callbacks?